### PR TITLE
Predictable ips

### PR DESCRIPTION
**Pull request by beagles**
Jira: https://issues.redhat.com/browse/OSPRH-6042

### CODE REVIEW

File: hack/configure_local_webhook.sh [hunk 1]
```diff
@@ -6,8 +6,8 @@ SKIP_CERT=${SKIP_CERT:-false}
 CRC_IP=${CRC_IP:-$(/sbin/ip -o -4 addr list crc | awk '{print $4}' | cut -d/ -f1)}
 
 #Open 9443
-sudo firewall-cmd --zone=libvirt --add-port=9443/tcp
-sudo firewall-cmd --runtime-to-permanent
+#sudo firewall-cmd --zone=libvirt --add-port=9443/tcp
+#sudo firewall-cmd --runtime-to-permanent
```
Comment (beagles) [inline hack/configure_local_webhook.sh:10]: Spotted... will update shortly to remove hack.

**Comment by beagles** [comment]
I still have some testing to do but this is ready for review

**Review by weinimo — CHANGES_REQUESTED** [review]

File: pkg/octavia/lb_mgmt_network.go [hunk 2]
```diff
@@ -897,3 +899,35 @@ func EnsureAmphoraManagementNetwork(
 		ManagementSubnetGateway: networkParameters.ProviderGateway.String(),
 	}, nil
 }
+
```
[the remark this answers is not in the document]
Comment (beagles) [reply pkg/octavia/lb_mgmt_network.go:902]: Agreed.

File: pkg/octavia/lb_mgmt_network.go [hunk 3]
```diff
@@ -897,3 +899,35 @@ func EnsureAmphoraManagementNetwork(
...
+		if !predParams.CIDR.Contains(endRange) {
+			return nil, fmt.Errorf("predictable IPs: cannot allocate %d IP addresses in %s", LbProvPredictablePoolSize, predParams.CIDR)
+		}
+		endRange = endRange.Next()
+	}
+	predParams.RangeEnd = endRange
+	return predParams, nil
+}
+
+func GetNextIP(predParams *NADIpam, currentValues map[string]bool) (string, error) {
```
[the remark this answers is not in the document]
Comment (beagles) [reply pkg/octavia/lb_mgmt_network.go:918]: Agreed.

File: templates/octaviaamphoracontroller/bin/setipalias.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+if not len(octavia_interface):
+    print('octavia network attachment not present')
+    sys.exit(1)
+
+# The file containing our IP alias has the worker node name for
+# a filename.
+
+nodefile = "hm_%s" % os.environ.get('NODE_NAME').strip()
+if nodefile:
```
[the remark this answers is not in the document]
Comment (beagles) [reply templates/octaviaamphoracontroller/bin/setipalias.py:23]: Also agreed. I'm amending the script to error out also if the conditions are not met. With predictable IPs these alias IPs are now essential.

**Pull request merged**